### ISSUE TITLE
Compile fails with c++11 disabled in LEDE 17.0.1

### ISSUE THREAD

**Issue by gilfrade** [mfontanini/libtins#282]
Hello, i tried to update the existing lede package (i'm not the maintainer) from 3.5 to 4.0 and several checks for C++11 failed with gcc 5.5 (should have support). Then i disabled C++11 support and it still fails because it still is trying to use c++11 features. What can i do?

# Build output:

Collecting package info: done
make[1]: Entering directory '/home/openwrt/source'
make[2]: Entering directory '/home/openwrt/source/package/libs/toolchain'
echo "libc" >> /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/pkginfo/toolchain.default.install
echo "libgcc" >> /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/pkginfo/toolchain.default.install
echo "libstdcpp" >> /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/pkginfo/toolchain.default.install
echo "libpthread" >> /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/pkginfo/toolchain.default.install
echo "librt" >> /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/pkginfo/toolchain.default.install
make[2]: Leaving directory '/home/openwrt/source/package/libs/toolchain'
make[2]: Entering directory '/home/openwrt/source/package/libs/libpcap'
echo "libpcap" >> /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/pkginfo/libpcap.default.install
make[2]: Leaving directory '/home/openwrt/source/package/libs/libpcap'
make[2]: Entering directory '/home/openwrt/source/feeds/packages/libs/libtins'
touch /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.prepared_fb435867b699bb66bb59cfeb006e016d_6664517399ebbbc92a37c5bb081b5c53_check
. /home/openwrt/source/include/shell.sh; xzcat /home/openwrt/source/dl/libtins-v4.0.tar.xz | tar -C /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.. -xf - 
[ ! -d ./src/ ] || cp -fpR ./src/* /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins
touch /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.prepared_fb435867b699bb66bb59cfeb006e016d_6664517399ebbbc92a37c5bb081b5c53
rm -f /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/stamp/.libtins_installed
Makefile:54: recipe for target '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.configured_68b329da9893e34099c7d8ad5cb9c940' failed
make[2]: [/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 123 (ignored)
mkdir -p /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins
(cd /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins; CFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins:libtins -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " CXXFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins:libtins -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " LDFLAGS="-L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/lib -L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/usr/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/lib -znow -zrelro " cmake -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_VERSION=1 -DCMAKE_SYSTEM_PROCESSOR=mipsel -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_C_COMPILER="/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-gcc" -DCMAKE_C_COMPILER_ARG1="" -DCMAKE_CXX_COMPILER="/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-g++" -DCMAKE_CXX_COMPILER_ARG1="" -DCMAKE_ASM_COMPILER="/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-gcc" -DCMAKE_ASM_COMPILER_ARG1="" -DCMAKE_EXE_LINKER_FLAGS:STRING="-L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/lib -L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/usr/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/lib -znow -zrelro" -DCMAKE_MODULE_LINKER_FLAGS:STRING="-L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/lib -L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/usr/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_SHARED_LINKER_FLAGS:STRING="-L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/lib -L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/usr/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_AR="/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-gcc-ar" -DCMAKE_NM="/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-gcc-nm" -DCMAKE_RANLIB="/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-gcc-ranlib" -DCMAKE_FIND_ROOT_PATH="/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr;/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl" -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_STRIP=: -DCMAKE_INSTALL_PREFIX=/usr -DDL_LIBRARY=/home/openwrt/source/staging_dir/target-mipsel_24kc_musl -DCMAKE_PREFIX_PATH=/home/openwrt/source/staging_dir/target-mipsel_24kc_musl -DCMAKE_SKIP_RPATH=TRUE -D_RUN_RESULT_VAR=FORCE -DLIBTINS_ENABLE_WPA2=0 -DLIBTINS_ENABLE_CXX11=0 -DLIBTINS_BUILD_EXAMPLES=0 -DLIBTINS_BUILD_TESTS=0 /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins )
-- The C compiler identification is GNU 5.5.0
-- The CXX compiler identification is GNU 5.5.0
-- Check for working C compiler: /home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-gcc
-- Check for working C compiler: /home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-g++
-- Check for working CXX compiler: /home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/bin/mipsel-openwrt-linux-musl-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using specified 'Release' build type.
-- Build will generate a shared library. Use LIBTINS_BUILD_SHARED=0 to perform a static build
-- Found PCAP: /home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/lib/libpcap.so  
-- Performing Test PCAP_LINKS_SOLO
-- Performing Test PCAP_LINKS_SOLO - Success
-- Looking for pcap_get_pfring_id
-- Looking for pcap_get_pfring_id - not found
-- Looking for pcap_set_immediate_mode
-- Looking for pcap_set_immediate_mode - found
-- Looking for pcap_set_tstamp_precision
-- Looking for pcap_set_tstamp_precision - found
-- Performing Test _HAS_CXX11_FLAG
-- Performing Test _HAS_CXX11_FLAG - Success
-- Checking C++ support for "decltype"
-- Checking C++ support for "decltype": not supported
-- Checking C++ support for "declval"
-- Checking C++ support for "declval": not supported
-- Checking C++ support for "initializer_list"
-- Checking C++ support for "initializer_list": not supported
-- Checking C++ support for "rvalue-references"
-- Checking C++ support for "rvalue-references": not supported
-- Checking C++ support for "functional"
-- Checking C++ support for "functional": not supported
-- Checking C++ support for "chrono"
-- Checking C++ support for "chrono": not supported
-- Checking C++ support for "noexcept"
-- Checking C++ support for "noexcept": not supported
-- Checking C++ support for "builtin-swap"
-- Checking C++ support for "builtin-swap": not supported
CMake Warning at CMakeLists.txt:117 (MESSAGE):
  Disabling C++11 features.  Use LIBTINS_ENABLE_CXX11=1 to enable them.
  Unless you are using an old compiler, you should enable this option, as it
  increases the library's performance

-- Enabling IEEE 802.11 support.
-- Disabling WPA2 decryption support.
-- Disabling TCPIP classes
-- Boost version: 1.65.1
-- Disabling ACK tracking support
-- Disabling TCP stream custom data support
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_ASM_COMPILER
    CMAKE_ASM_COMPILER_ARG1
    CMAKE_EXE_LINKER_FLAGS
    CMAKE_MODULE_LINKER_FLAGS
    DL_LIBRARY

-- Build files have been written to: /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins
rm -f /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.configured_*
touch /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.configured_68b329da9893e34099c7d8ad5cb9c940
rm -f /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.built
touch /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.built_check
CFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins:libtins -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/include -I/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/include -I/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/usr/include -I/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/include/fortify -I/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/include " CXXFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins:libtins -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/include -I/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/include -I/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/usr/include -I/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/include/fortify -I/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/include " LDFLAGS="-L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/usr/lib -L/home/openwrt/source/staging_dir/target-mipsel_24kc_musl/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/usr/lib -L/home/openwrt/source/staging_dir/toolchain-mipsel_24kc_gcc-5.5.0_musl/lib -znow -zrelro " make  -C /home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/. AR="mipsel-openwrt-linux-musl-gcc-ar" AS="mipsel-openwrt-linux-musl-gcc -c -Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -iremap/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins:libtins -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro" LD=mipsel-openwrt-linux-musl-ld NM="mipsel-openwrt-linux-musl-gcc-nm" CC="mipsel-openwrt-linux-musl-gcc" GCC="mipsel-openwrt-linux-musl-gcc" CXX="mipsel-openwrt-linux-musl-g++" RANLIB="mipsel-openwrt-linux-musl-gcc-ranlib" STRIP=mipsel-openwrt-linux-musl-strip OBJCOPY=mipsel-openwrt-linux-musl-objcopy OBJDUMP=mipsel-openwrt-linux-musl-objdump SIZE=mipsel-openwrt-linux-musl-size CROSS="mipsel-openwrt-linux-musl-" ARCH="mipsel" CMAKE_COMMAND='/home/openwrt/source/staging_dir/host/bin/cmake' CMAKE_DISABLE_cmake_check_build_system=1 ;
make[3]: Entering directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
make[4]: Entering directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
make[5]: Entering directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
Scanning dependencies of target tins
make[5]: Leaving directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
make[5]: Entering directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
[  1%] Building CXX object src/CMakeFiles/tins.dir/address_range.cpp.o
[  2%] Building CXX object src/CMakeFiles/tins.dir/arp.cpp.o
[  4%] Building CXX object src/CMakeFiles/tins.dir/bootp.cpp.o
[  5%] Building CXX object src/CMakeFiles/tins.dir/crypto.cpp.o
[  6%] Building CXX object src/CMakeFiles/tins.dir/detail/address_helpers.cpp.o
[  8%] Building CXX object src/CMakeFiles/tins.dir/detail/icmp_extension_helpers.cpp.o
[  9%] Building CXX object src/CMakeFiles/tins.dir/detail/pdu_helpers.cpp.o
[ 10%] Building CXX object src/CMakeFiles/tins.dir/detail/sequence_number_helpers.cpp.o
[ 12%] Building CXX object src/CMakeFiles/tins.dir/dhcp.cpp.o
[ 13%] Building CXX object src/CMakeFiles/tins.dir/dhcpv6.cpp.o
[ 15%] Building CXX object src/CMakeFiles/tins.dir/dns.cpp.o
[ 16%] Building CXX object src/CMakeFiles/tins.dir/dot3.cpp.o
[ 17%] Building CXX object src/CMakeFiles/tins.dir/dot1q.cpp.o
[ 19%] Building CXX object src/CMakeFiles/tins.dir/eapol.cpp.o
[ 20%] Building CXX object src/CMakeFiles/tins.dir/ethernetII.cpp.o
[ 21%] Building CXX object src/CMakeFiles/tins.dir/handshake_capturer.cpp.o
[ 23%] Building CXX object src/CMakeFiles/tins.dir/hw_address.cpp.o
[ 24%] Building CXX object src/CMakeFiles/tins.dir/icmp_extension.cpp.o
[ 26%] Building CXX object src/CMakeFiles/tins.dir/icmp.cpp.o
[ 27%] Building CXX object src/CMakeFiles/tins.dir/icmpv6.cpp.o
[ 28%] Building CXX object src/CMakeFiles/tins.dir/ip_reassembler.cpp.o
[ 30%] Building CXX object src/CMakeFiles/tins.dir/ip.cpp.o
[ 31%] Building CXX object src/CMakeFiles/tins.dir/ip_address.cpp.o
[ 32%] Building CXX object src/CMakeFiles/tins.dir/ipv6.cpp.o
[ 34%] Building CXX object src/CMakeFiles/tins.dir/ipv6_address.cpp.o
[ 35%] Building CXX object src/CMakeFiles/tins.dir/ipsec.cpp.o
[ 36%] Building CXX object src/CMakeFiles/tins.dir/llc.cpp.o
[ 38%] Building CXX object src/CMakeFiles/tins.dir/loopback.cpp.o
[ 39%] Building CXX object src/CMakeFiles/tins.dir/mpls.cpp.o
[ 41%] Building CXX object src/CMakeFiles/tins.dir/memory_helpers.cpp.o
[ 42%] Building CXX object src/CMakeFiles/tins.dir/network_interface.cpp.o
[ 43%] Building CXX object src/CMakeFiles/tins.dir/packet_sender.cpp.o
[ 45%] Building CXX object src/CMakeFiles/tins.dir/pdu.cpp.o
[ 46%] Building CXX object src/CMakeFiles/tins.dir/pdu_iterator.cpp.o
[ 47%] Building CXX object src/CMakeFiles/tins.dir/pdu_option.cpp.o
[ 49%] Building CXX object src/CMakeFiles/tins.dir/pppoe.cpp.o
[ 50%] Building CXX object src/CMakeFiles/tins.dir/radiotap.cpp.o
[ 52%] Building CXX object src/CMakeFiles/tins.dir/rawpdu.cpp.o
[ 53%] Building CXX object src/CMakeFiles/tins.dir/rsn_information.cpp.o
[ 54%] Building CXX object src/CMakeFiles/tins.dir/sll.cpp.o
[ 56%] Building CXX object src/CMakeFiles/tins.dir/snap.cpp.o
[ 57%] Building CXX object src/CMakeFiles/tins.dir/stp.cpp.o
[ 58%] Building CXX object src/CMakeFiles/tins.dir/tcp.cpp.o
[ 60%] Building CXX object src/CMakeFiles/tins.dir/tcp_ip/ack_tracker.cpp.o
[ 61%] Building CXX object src/CMakeFiles/tins.dir/tcp_ip/flow.cpp.o
[ 63%] Building CXX object src/CMakeFiles/tins.dir/tcp_ip/data_tracker.cpp.o
[ 64%] Building CXX object src/CMakeFiles/tins.dir/tcp_ip/stream.cpp.o
[ 65%] Building CXX object src/CMakeFiles/tins.dir/tcp_ip/stream_follower.cpp.o
[ 67%] Building CXX object src/CMakeFiles/tins.dir/tcp_ip/stream_identifier.cpp.o
[ 68%] Building CXX object src/CMakeFiles/tins.dir/timestamp.cpp.o
[ 69%] Building CXX object src/CMakeFiles/tins.dir/udp.cpp.o
[ 71%] Building CXX object src/CMakeFiles/tins.dir/utils/checksum_utils.cpp.o
[ 72%] Building CXX object src/CMakeFiles/tins.dir/utils/frequency_utils.cpp.o
[ 73%] Building CXX object src/CMakeFiles/tins.dir/utils/radiotap_parser.cpp.o
[ 75%] Building CXX object src/CMakeFiles/tins.dir/utils/radiotap_writer.cpp.o
[ 76%] Building CXX object src/CMakeFiles/tins.dir/utils/routing_utils.cpp.o
[ 78%] Building CXX object src/CMakeFiles/tins.dir/utils/resolve_utils.cpp.o
[ 79%] Building CXX object src/CMakeFiles/tins.dir/utils/pdu_utils.cpp.o
[ 80%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_base.cpp.o
[ 82%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_data.cpp.o
[ 83%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_mgmt.cpp.o
[ 84%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_beacon.cpp.o
[ 86%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_assoc.cpp.o
[ 87%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_auth.cpp.o
[ 89%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_probe.cpp.o
[ 90%] Building CXX object src/CMakeFiles/tins.dir/dot11/dot11_control.cpp.o
[ 91%] Building CXX object src/CMakeFiles/tins.dir/sniffer.cpp.o
/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/src/sniffer.cpp:227:5: warning: identifier 'nullptr' is a keyword in C++11 [-Wc++0x-compat]
     if (method == nullptr) {
     ^
/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/src/sniffer.cpp: In member function 'void Tins::BaseSniffer::set_pcap_sniffing_method(Tins::BaseSniffer::PcapSniffingMethod)':
/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/src/sniffer.cpp:227:19: error: 'nullptr' was not declared in this scope
     if (method == nullptr) {
                   ^
src/CMakeFiles/tins.dir/build.make:1646: recipe for target 'src/CMakeFiles/tins.dir/sniffer.cpp.o' failed
make[5]: *** [src/CMakeFiles/tins.dir/sniffer.cpp.o] Error 1
make[5]: Leaving directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
CMakeFiles/Makefile2:117: recipe for target 'src/CMakeFiles/tins.dir/all' failed
make[4]: *** [src/CMakeFiles/tins.dir/all] Error 2
make[4]: Leaving directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
Makefile:129: recipe for target 'all' failed
make[3]: *** [all] Error 2
make[3]: Leaving directory '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins'
Makefile:54: recipe for target '/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.built' failed
make[2]: *** [/home/openwrt/source/build_dir/target-mipsel_24kc_musl/libtins/.built] Error 2
make[2]: Leaving directory '/home/openwrt/source/feeds/packages/libs/libtins'
package/Makefile:109: recipe for target 'package/feeds/packages/libtins/compile' failed
make[1]: *** [package/feeds/packages/libtins/compile] Error 2
make[1]: Leaving directory '/home/openwrt/source'
/home/openwrt/source/include/toplevel.mk:207: recipe for target 'package/feeds/packages/libtins/compile' failed
make: *** [package/feeds/packages/libtins/compile] Error 2

# LEDE Makefile

include $(TOPDIR)/rules.mk

PKG_NAME:=libtins
PKG_RELEASE:=1

PKG_MAINTAINER:= Steven Hessing <steven.hessing@gmail.com>

PKG_SOURCE_PROTO:=git
PKG_SOURCE_URL:=https://github.com/mfontanini/libtins.git
PKG_SOURCE_VERSION:=v4.0
PKG_SOURCE_SUBDIR:=$(PKG_NAME)
PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.xz
PKG_MIRROR_HASH:=8fd25e23a634d2e98ab423d011733d0c3b753900

PKG_LICENSE:=BSD-2-Clause
PKG_LICENSE_FILES:=LICENSE

PKG_BUILD_DIR := $(BUILD_DIR)/$(PKG_NAME)
PKG_BUILD_PARALLEL:=1

CMAKE_INSTALL:=1
include $(INCLUDE_DIR)/package.mk
include $(INCLUDE_DIR)/cmake.mk
CMAKE_OPTIONS += -D_RUN_RESULT_VAR=FORCE
CMAKE_OPTIONS += -DLIBTINS_ENABLE_WPA2=0
CMAKE_OPTIONS += -DLIBTINS_ENABLE_CXX11=0
CMAKE_OPTIONS += -DLIBTINS_BUILD_EXAMPLES=0
CMAKE_OPTIONS += -DLIBTINS_BUILD_TESTS=0

define Package/libtins
	SECTION:=net
	CATEGORY:=Libraries
	TITLE:=libtins 
	URL:=http://libtins.github.io/
	DEPENDS:=+libstdcpp +libpcap 
endef

define Package/libtins/description
libtins is a high-level, multiplatform C++ network packet sniffing and crafting library.
endef

define Package/libtins/install
    $(INSTALL_DIR) $(1)/usr/lib
	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/lib/libtins.so.4.0 $(1)/usr/lib/
endef

$(eval $(call BuildPackage,libtins))

**Comment by mfontanini** [comment]
Sorry, I missed that `nullptr`, it shouldn't have been there. Can you pull from master? The on C++11 build should work now. Still, it's odd gcc 5+ doesn't build this.

**Comment by gilfrade** [comment]
With the last master commit it builds fine without c++11 suport, its fxed!. I also edited the c++11 checks from cmake and it build fine with c++11 on musl LEDE. Tried a quick example and worked.
Maybe the tests checks fails on musl and it misleads cmake to think it does not have support?

**Comment by mfontanini** [comment]
Hmm interesting. You're cross compiling this, right? It may somehow be calling the toolchain incorrectly on the CMake tests. Last time I tried debugging an issue on this I had to modify `cmake/Modules/CheckCXXFeatures.cmake` and use `OUTPUT_VARIABLE` on [try_compile](https://cmake.org/cmake/help/v3.0/command/try_compile.html) to see what's actually happening under the hood IIRC. I don't really know of a better way to do so unfortunately.
